### PR TITLE
Fix Initial Sync Stall When User Switches Apps During Sync

### LINKED ISSUE #3341: Initial Sync stalls when app is running in the background
***Apk Vesrion*** - `opensrp-dc-clinic-v0.0.5.apk`
**Device**: `SM-T380`
***Android Version*** - `9`

**Issue Description**
Initial Sync is stalling when the app is pushed to the background. On reopening the app, sync does not progress to completion. It remains stuck at the percentage it was at when the user switched to another app. There are no errors recorded in the logs

**Steps to reproduce**
1. Clear the app's data
2. Initialize a new login session by entering the app ID
3. Log into the app and Set Pin
4. After initial sync starts, switch to a different app

**Expected Results**

- [ ] Initial sync should continue running when the app is pushed to the background until completion

### PR DESCRIPTION
**Pull request by qiarie**
- Check sync worker status when loading register & geowidget fragments

**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #3341 

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainViewModel.kt [hunk 1]
```diff
@@ -267,9 +269,21 @@ constructor(
...
   /** This function is used to schedule tasks that are intended to run periodically */
   fun schedulePeriodicJobs() {
     workManager.run {
+      schedulePeriodically<AppSyncWorker>(
+        workId = AppSyncWorker.WORK_ID,
+        repeatInterval = applicationConfiguration.syncInterval,
+        timeUnit = TimeUnit.MINUTES,
+        requiresNetwork = true,
+        initialDelay = applicationConfiguration.syncInterval,
+      )
```
Comment (ellykits) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainViewModel.kt:285]: There are instances where we would not want to schedule sync until locations for syncing data are selected. This is a use case for EUSM project, which is why the sync job was not included in this method.

Comment (qiarie) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainViewModel.kt:285]: Noted. Let me review this

Comment (ellykits) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainViewModel.kt:286]: Also, how do we react to the sync statuses using this approach? We need to collect the flow items. That is why the functionality is in the `SyncBroadcaster` class.

Comment (ellykits) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainViewModel.kt:286]: To ensure that all sync listeners are notified.

**Comment by qiarie** [comment]
Resolved in #3467

**Pull request closed without merging**